engine/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/DockerServiceImpl.java

```diff
@@ -0,0 +1,202 @@
+package com.odysseusinc.arachne.executionengine.service.impl;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.async.ResultCallback;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.model.Frame;
+import com.github.dockerjava.api.model.Image;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient;
+import com.github.dockerjava.transport.DockerHttpClient;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.AnalysisResultStatusDTO;
+import com.odysseusinc.arachne.execution_engine_common.api.v1.dto.AnalysisSyncRequestDTO;
+import com.odysseusinc.arachne.executionengine.config.properties.DockerProperties;
+import com.odysseusinc.arachne.executionengine.model.descriptor.DescriptorBundle;
+import com.odysseusinc.arachne.executionengine.service.DockerService;
+import com.odysseusinc.arachne.executionengine.util.AnalysisCallback;
+import com.odysseusinc.datasourcemanager.krblogin.KrbConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+@Service
+@ConditionalOnProperty(name = "use.docker", havingValue = "true")
+public class DockerServiceImpl implements DockerService
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DockerServiceImpl.class);
+    private final ThreadPoolTaskExecutor taskExecutor;
+    private final DockerProperties dockerProperties;
+
+    @Autowired
+    public DockerServiceImpl(ThreadPoolTaskExecutor taskExecutor, DockerProperties dockerProperties) {
+        this.taskExecutor = taskExecutor;
+        this.dockerProperties = dockerProperties;
+    }
+
+    public DockerClient dockerClient() {
+        DefaultDockerClientConfig.Builder builder = DefaultDockerClientConfig.createDefaultConfigBuilder()
+                .withDockerTlsVerify(false);
+        if (dockerProperties.getRegistryUrl() != null) {
+            builder.withRegistryUrl(dockerProperties.getRegistryUrl());
+        }
+        if (dockerProperties.getRegistryPassword() != null) {
```

Comment:
Minor: `Optional.ofNullable(dockerProperties.getRegistryUrl()).ifPresent(builder::withRegistryUrl)` plays nicely in such situations